src/lib/ppx_coda/bin_io_unversioned.ml

```diff
@@ -0,0 +1,101 @@
+(* bin_io_unversioned *)
+
+(* use [@@deriving bin_io_unversioned] for serialized types that
+   are not send to nodes or persisted, namely the daemon RPC
+   types for communication between the client and daemon, which
+   are known to be running the same version of the software.
+
+   The deriver here calls the derivers for bin_io after the lint
+   phase. We avoid linter errors that would occur if
+   we used [@@deriving bin_io] directly.
+*)
+
+open Core_kernel
+open Ppxlib
+open Versioned_util
+
+let deriver = "bin_io_unversioned"
+
+let bin_io_gens :
+    (   ctxt:Expansion_context.Deriver.t
+     -> rec_flag * type_declaration list
+     -> (string * expression) list
+     -> structure_item list)
+    list =
+  List.map ["bin_shape"; "bin_read"; "bin_write"; "bin_type_class"]
+    ~f:(fun name ->
+      let deriver = Option.value_exn (Ppx_derivers.lookup name) in
+      (* deriver is an instance of Ppxlib.Deriving.Deriver.deriver
+
+           if instead we consider it to be an instance of Ppx_deriving.t, the 0th
+           field, which should be the name, but prints out as garbage
+
+           the 0th field of deriver is a string "Ppxlib__Deriving.Deriver.T", perhaps an artifact
+             of the += type extension
+           the 1st field of deriver appears to be of type Ppxlib.Deriving.Deriver.t, and the 0th field
+             of that gives us an Actual_deriver.t
+
+           the 0th field of the Actual_deriver.t should be the name, which prints out as
+             expected
+
+           the number of fields of the Actual_deriver.t is 8, which corresponds to the type
+             definition
+
+           from the Actual_deriver.t, we pull out the 1st field, the Generator.t with the label
+             str_type_decl; we then apply the generator
+        *)
+      let actual_deriver = Obj.(obj (field (field (repr deriver) 1) 0)) in
```

Comment:
Nit: This might be clearer as pipes and named indexes, e.g.
```ocaml
let actual_deriver =
  Obj.
    (repr deriver
       (* Ppxlib.Deriving.T x -> x *)
      |> field extension_constructor_argument_idx
       (* Ppxlib.Deriving.Actual_deriver x -> x *)
      |> field constructor_argument_idx)
in
```